sdk/core/Azure.Core.Experimental/src/Serialization/JsonObjectSerializer.cs

```diff
@@ -59,5 +67,85 @@ public override async ValueTask<object> DeserializeAsync(Stream stream, Type ret
         {
             return await JsonSerializer.DeserializeAsync(stream, returnType, _options, cancellationToken).ConfigureAwait(false);
         }
+
+        /// <inheritdoc/>
+        public string? GetSerializedMemberName(MemberInfo memberInfo)
+        {
+            Argument.AssertNotNull(memberInfo, nameof(memberInfo));
+
+            if (!_cache.TryGetValue(memberInfo.ReflectedType, out MemberInfo[] members))
+            {
+                members = GetMembers(memberInfo.ReflectedType).ToArray();
+            }
+
+            foreach (MemberInfo member in members)
+            {
+                if (member == memberInfo)
+                {
+                    return GetPropertyName(memberInfo);
```

Comment:
Looks like this method call is all we need. Why do we need to re-walk the list every time?

Reply:
Because the method may be ignored. `GetMembers` mimics what STJ does. I can cache the property names to make subsequent calls faster, but the initial looping is necessary.

Reply:
We can have this logic in GetSerializedName

```
   if (propertyInfo.GetMethod?.IsPublic == true ||
                    propertyInfo.SetMethod?.IsPublic == true)
                {
                    if (propertyInfo.GetCustomAttribute<JsonIgnoreAttribute>() != null)
                    {
                        continue;
                    }

                    // Ignore - but do not assert correctness - for JsonExtensionDataAttribute based on
                    // https://github.com/dotnet/corefx/blob/v3.1.0/src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs#L244-L261
                    if (propertyInfo.GetCustomAttribute<JsonExtensionDataAttribute>() != null)
                    {
                        continue;
                    }

                    // No need to validate collisions since they are based on the serialized name.
                    yield return propertyInfo;
                }
```